org.lflang/src/org/lflang/generator/c/CModesGenerator.java

```diff
@@ -31,18 +31,18 @@ public static void generateDeclarations(
         if (!allModes.isEmpty()) {
             // Reactor's mode instances and its state.
             body.pr(String.join("\n", 
-                "reactor_mode_t _lf__modes["+reactor.getModes().size()+"];",
-                "reactor_mode_state_t _lf__mode_state;"
```

Comment:
> Is there a benefit in doing it this way?

We needed this because we wanted to call `_LF_ SET_MODE` in the `.set` method of the `mode` type in the Python target. `_LF_ SET_MODE` expects a `self` pointer to a self struct that has a `reactor_mode_state_t _lf__mode_state;` member. The idea is that we want to re-use `_LF_ SET_MODE` so that any changes in the C runtime to the functionality of this macro is also automatically reflected in the Python runtime.

The implementation of the `.set` method is [here](https://github.com/lf-lang/reactor-c-py/blob/78dc0b241d2797423d824080df7e7f685fd45b30/lib/modal_models/impl.c#L40) in case you are interested.

Reply:
That makes sense. Thanks.